sdk/storage/inc/common/file_io.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#ifdef _WIN32
+#include <Windows.h>
+#endif
+
+#include <cstdint>
+#include <string>
+
+namespace Azure { namespace Storage { namespace Details {
+
+#ifdef _WIN32
+  using FileHandle = HANDLE;
+#else
+  using FileHandle = int;
+#endif
+
+  class FileReader {
+  public:
+    FileReader(const std::string& filename);
+
+    ~FileReader();
+
```

Comment:
FileReader does not provide `Read` is somewhat strange. However, I don't find a place to call this `Read`. It is your choice to or not to add the API.

Reply:
okay